tests/integration-tests/tests/update/test_update.py

```diff
@@ -190,6 +192,42 @@ def test_update_slurm(region, pcluster_config_reader, s3_bucket_factory, cluster
     # - Test extra json
 
 
+@pytest.mark.dimensions("us-west-1", "c5.xlarge", "*", "slurm")
+@pytest.mark.usefixtures("instance", "scheduler")
+@pytest.mark.parametrize(
+    "imds_secured, imds_secured_updated, users_allow_list",
+    [
+        (True, False, {"root": True, "pcluster": True, "slurm": True}),
+        (False, True, {"root": True, "pcluster": True, "slurm": False}),
+    ],
+)
+def test_update_imds_secured(
+    imds_secured,
+    imds_secured_updated,
+    users_allow_list,
+    region,
+    os,
+    pcluster_config_reader,
+    clusters_factory,
+    architecture,
+    test_datadir,
+):
+    """Test IMDS access with different configurations"""
+    custom_ami = retrieve_latest_ami(region, os, ami_type="pcluster", architecture=architecture)
+    cluster_config = pcluster_config_reader(custom_ami=custom_ami, imds_secured=imds_secured)
+    cluster = clusters_factory(cluster_config, raise_on_error=False)
+
+    assert_head_node_is_running(region, cluster)
+
+    cluster_config_updated = pcluster_config_reader(
+        config_file="pcluster.config.update.yaml", custom_ami=custom_ami, imds_secured=imds_secured_updated
```

Comment:
the templates that are used to generate the initial and updated configuration files look to be the same. can the update config be dropped?

Reply:
I tried, but at the end I discovered that you must use two distinct files due to the current implementation of `pcluster_config_reader`. 
I think `pcluster_config_reader` should be changed accordingly, but since it is out of scope for this PR, I'd stay with this standard approach.